sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/GroupingSqlAggregator.java

```diff
@@ -90,7 +91,17 @@ public Aggregation toDruidAggregation(
         }
       }
     }
-    AggregatorFactory factory = new GroupingAggregatorFactory(name, arguments);
+    AggregatorFactory factory;
+    try {
+      factory = new GroupingAggregatorFactory(name, arguments);
+    }
+    catch (Exception e) {
```

Comment:
pass by comment, is this the correct way to handle any errors coming out of the agg factory?
A couple of things that I find ambiguous : 
1. `Grouping Aggregation [%s] is not supported` seems like misleading message incase there's an actual bug in the agg factory
2. `CannotBuildQueryException` only takes in the message of the exception, which would make things harder to debug without a stack if there's an exception.

Reply:
@rohangarg Thanks for the comment.
I have updated the exception message to remove not supported. I think the exception here would only be caused by the Precondition checks and that could be easily understood by the exception message
Made the exception look similar to the one we have in `EarliestLatestBySqlAggregator` and return null.
anyways, In the end `DruidQuery` would throw `CannotBuildQueryException`. So it made sense to not throw in the aggregator class.

Does this look good to you now?

Reply:
Thanks for the fixups :)
A couple of things I would check are : 
1. How does the whole user message look when the exception is thrown - does that make sense?
2. Are all the exceptions thrown by the constructor distinguishable via the message - so that we aren't left with ambiguity when an exception arrives?

If both those things look fine, then it is good to go from my side too

Reply:
1. These exceptions which would be thrown here would result in not being able to plan the query in the current path and look for other ways. The custom rule which is introduced in this PR will make sure that the duplicate grouping would be removed and plan the query properly. In my understanding, the user would face this exception if the query  can not be planned at all. As part of this testcase which I have added, Initially this would fail in plan and then turns towards the rule, gets converted and generated the final plan.
2. By setting the planning error with the exception being received, this should be able to distinguish between the different exceptions based on the message

Reply:
Ah, for 1, I was actually mentioning the meaningfulness of planning error message which would get surfaced if a query can't get planned. It does look reasonable to me.
For 2, thanks for the clarification. 
LGTM